components/resourceDetails/index.test.jsx

```diff
@@ -70,6 +71,27 @@ describe("Resource details", () => {
     expect(asFragment()).toMatchSnapshot();
   });
 
+  it("copies resource address to the clipboard when the copy link button is clicked", async () => {
+    const writeText = jest.fn();
+    Object.assign(navigator, {
+      clipboard: {
+        writeText,
+      },
+    });
+
+    const { getByTestId } = renderWithTheme(
+      <DatasetProvider solidDataset={dataset}>
+        <ResourceDetails />
+      </DatasetProvider>
+    );
+    const copyLink = screen.getByTestId(TESTCAFE_RESOURCE_ADDRESS_TOOLTIP);
+    userEvent.click(copyLink);
+    await waitFor(() => {
+      expect(copyLink).toBeInTheDocument();
+      expect(writeText).toHaveBeenCalled();
+    });
```

Comment:
Generally multiple asserts in a waitFor is a bad practice, but until you can migrate to userEvebts from testing-lib you're kinda stuck with events being dispatched async from test execution.

With userEvent, you can just do: 

```
await user.click(copyLink);

expect(...)
exoect(...)
```